pkg/operator/controllers/checkers/ingresscertificatechecker/checker_test.go

```diff
@@ -31,37 +34,101 @@ func TestCheck(t *testing.T) {
 		name              string
 		ingressController *operatorv1.IngressController
 		clusterVersion    *configv1.ClusterVersion
+		cluster           *arov1alpha1.Cluster
 		wantErr           string
 	}{
 		{
-			name:           "default certificate is set",
+			name:    "shoud return clusterVersion not found error when ClusterVersion is nil",
+			wantErr: `clusterversions.config.openshift.io "version" not found`,
+		},
+		{
+			name:              "shoud return IngressController not found error when IngressController is nil",
+			ingressController: nil,
+			clusterVersion:    clusterVersion,
+			wantErr:           `ingresscontrollers.operator.openshift.io "default" not found`,
+		},
+		{
+			name:              "should return Cluster not found error when Cluster is nil",
+			clusterVersion:    clusterVersion,
+			ingressController: fakeIngressController(nil),
+			cluster:           nil,
+			wantErr:           "clusters.aro.openshift.io \"cluster\" not found",
+		},
+		{
+			name:              "should return certificate error (ARO's fault) when no certificate is set and we have a managed domain",
+			clusterVersion:    clusterVersion,
+			ingressController: fakeIngressController(nil),
+			cluster:           fakeCluster("whatever.aroapp.io"),
+			wantErr:           "missing ingress certificate for cluster with managed domain",
```

Comment:
would it make sense to use `errVar.Error()` instead of retyping it? would prevent us to have to change the tests if we want to change the error messages

Reply:
Yes! I 100% agree. I prefer sentinel errors. I was just following the tendency in the repo for consistency 😃 
Tks!